packages/api/src/target-group/target-groups.service.ts

```diff
@@ -37,16 +37,23 @@ export class TargetGroupsService {
     ): boolean {
         if (filters && contactAttributes) {
             for (const [key, value] of Object.entries(filters)) {
-                if (!value) continue;
+                if (!value || value.length === 0) continue;
+
+                if (Array.isArray(contactAttributes[key])) {
+                    for (const attribute of contactAttributes[key]) {
+                        if (value.includes(attribute)) {
+                            return true;
```

Comment:
You shouldn't return true in this case. Then the other attributes that come after the array will not be checked anymore. 
Example filters (note the order of the filters):
```
BRANCH -> [NEWS] 
SALUTATION -> FEMALE 
```
This would return true for contact since you instantly return true when checking the branch array (which is wrong)
```
BRANCH -> NEWS 
SALUTATION -> MALE
``` 
you should only skip the remaining attributes in the for loop (line 43) in this case.